client/rest-high-level/src/main/java/org/elasticsearch/client/Request.java

```diff
@@ -88,16 +90,76 @@ static Request bulk(BulkRequest bulkRequest) throws IOException {
         parameters.withTimeout(bulkRequest.timeout());
         parameters.withRefreshPolicy(bulkRequest.getRefreshPolicy());
 
-        // Bulk API only supports newline delimited JSON
-        XContentType xContentType = XContentType.JSON;
-        byte separator = xContentType.xContent().streamSeparator();
-        ContentType requestContentType = ContentType.create("application/x-ndjson");
+        // Bulk API only supports newline delimited JSON or Smile. Before executing
+        // the bulk, we need to check that all requests have the same content-type
+        // and this content-type is supported by the Bulk API.
+        List<XContentType> allowedContentTypes = Arrays.asList(XContentType.JSON, XContentType.SMILE);
```

Comment:
add a short method that checks the content type and get rid of the list? If this class becomes too big I am good with getting the bulk part out. some of this methods are going to be useful for msearch too at some point.